int32 agentId
int32 x
int32 y
int32 heading
int32[] responses